common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/model/codegen/BeanGenerator.java

```diff
@@ -0,0 +1,284 @@
+        if (property.getName().startsWith("/")) {
+            final TypeName valueType = getTypeNameSwitch().doSwitch(property.getType());
+            return FieldSpec.builder(
+                    ParameterizedTypeName.get(ClassName.get(Map.class), ClassName.get(String.class), valueType), "values", Modifier.PRIVATE)
+                    .build();
+        } else {
+            final TypeName valueType = getTypeNameSwitch().doSwitch(property.getType());
+            return FieldSpec.builder(valueType, property.getName(), Modifier.PRIVATE).build();
+        }
+    }
+
+    private MethodSpec getFieldSetter(final Property property) {
+        final FieldSpec fieldSpec = getFieldSpec(property);
+        final MethodSpec.Builder methodSpec;
+
+
+        if (property.getName().startsWith("/")) {
+            methodSpec = MethodSpec.methodBuilder("setValues");
+        } else {
+            methodSpec = MethodSpec.methodBuilder("set" + getPropertyMethodNameSuffix(property));
+        }
+
+        methodSpec.addParameter(fieldSpec.type, fieldSpec.name)
+                .addModifiers(Modifier.PUBLIC)
+                .addCode("this." + fieldSpec.name + " = " + fieldSpec.name + ";\n")
+                .build();
+        return methodSpec.build();
+    }
+
+    private MethodSpec getFieldGetter(final Property property) {
+        FieldSpec fieldSpec = getFieldSpec(property);
+        final MethodSpec.Builder methdBuilder;
+        if (property.getName().startsWith("/")) {
+            methdBuilder = MethodSpec.methodBuilder("values");
+            methdBuilder.addAnnotation(JsonAnySetter.class);
+            methdBuilder.addAnnotation(JsonAnyGetter.class);
+        } else {
+            final String attributePrefix = isBoolean(fieldSpec) ? "is" : "get";
+            methdBuilder = MethodSpec.methodBuilder(attributePrefix + getPropertyMethodNameSuffix(property))
+                    .addAnnotation(AnnotationSpec.builder(JsonProperty.class)
+                            .addMember("value", "$S", property.getName()).build());
+        }
+        methdBuilder.addModifiers(Modifier.PUBLIC)
+                .returns(fieldSpec.type)
+                .addCode("return " + fieldSpec.name + ";\n");
+        return methdBuilder.build();
+    }
+
+
+    private static String getPropertyMethodNameSuffix(final Property property) {
+        final String javaPropertyName = Arrays.stream(property.getName().split("_"))
+                .map(StringUtils::capitalize)
+                .collect(Collectors.joining());
+
+        return javaPropertyName;
+    }
+
+    private static boolean isBoolean(FieldSpec fieldSpec) {
+        return fieldSpec.type.equals(ClassName.BOOLEAN.box()) || fieldSpec.type.equals(ClassName.BOOLEAN);
+    }
+
+    public Path getPath(JavaFileObject javaFile, Path outputFolder) {
+        return Paths.get(outputFolder.toString(), javaFile.getName());
+    }
+
+    protected Stream<AnnotationSpec> getDiscriminatorAnnotations(final ObjectType object) {
```

Comment:
Here we should rename the parameter to `objectType` too.

Reply:
nice catch 👍